File: sdks/java/basys.sdk/src/main/java/org/eclipse/basyx/aas/metamodel/hashmap/aas/qualifier/HasDataSpecification.java
package org.eclipse.basyx.aas.metamodel.hashmap.aas.qualifier;

import java.util.HashMap;
import java.util.HashSet;

import org.eclipse.basyx.aas.api.metamodel.aas.qualifier.IHasDataSpecification;
import org.eclipse.basyx.aas.api.metamodel.aas.reference.IReference;
import org.eclipse.basyx.aas.metamodel.facades.HasDataSpecificationFacade;
import org.eclipse.basyx.aas.metamodel.hashmap.aas.reference.Reference;

/**
 * HasDataSpecification class
 * 
 * @author elsheikh, schnicke
 *
 */
public class HasDataSpecification extends HashMap<String, Object> implements IHasDataSpecification {
	
	public static final String HASDATASPECIFICATION="hasDataSpecification";

	/**
	 * Version of serialized instances
	 */
	private static final long serialVersionUID = 1L;

	/**
	 * Constructor
	 */
	public HasDataSpecification() {
		// Default values
		put(HASDATASPECIFICATION, new HashSet<Reference>());
	}

	@Override
	public HashSet<IReference> getDataSpecificationReferences() {
		return new HasDataSpecificationFacade(this).getDataSpecificationReferences();
	}

	@Override
	public void setDataSpecificationReferences(HashSet<IReference> ref) {
		new HasDataSpecificationFacade(this).setDataSpecificationReferences(ref);
		
	}
}

File: sdks/c++/basys.sdk.cc/src/aas/impl/metamodel/hashmap/aas/qualifier/HasDataSpecification.cpp
/*
 * HasDataSpecification.cpp
 *
 *      Author: wendel
 */

#include "HasDataSpecification.h"

namespace basyx {
namespace submodel {
namespace metamodel {
namespace map {
namespace qualifier {

using namespace aas::reference;

basyx::specificCollection_t<IReference> HasDataSpecification::getDataSpecificationReferences() const
{
  return this->references;
}

void HasDataSpecification::setDataSpecificationReferences(const basyx::specificCollection_t<aas::reference::IReference> & references)
{
  this->references = references;
}

}
}
}
}
}

